File: schemas/course.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any
from datetime import datetime

class QuizBase(BaseModel):
    id: str
    question: str
    options: List[str]
    correctOption: int

class QuizCreate(QuizBase):
    pass

class QuizResponse(QuizBase):
    id: str

    class Config:
        from_attributes = True

class ChapterBase(BaseModel):
    id: str
    title: str
    content: str

class ChapterCreate(ChapterBase):
    quiz: List[QuizCreate]

class ChapterResponse(ChapterBase):
    id: str
    quiz: List[QuizResponse]
    completed: Optional[bool] = False

    class Config:
        from_attributes = True

class CourseBase(BaseModel):
    title: str
    description: str
    imageUrl: str

class CourseCreate(CourseBase):
    chapters: List[ChapterCreate]

class CourseUpdate(CourseBase):
    chapters: List[ChapterCreate]

class CourseResponse(CourseBase):
    id: str
    chapters: List[ChapterResponse]
    progress: Optional[int] = 0
    enrolled: Optional[bool] = False
    enrollmentCode: Optional[str] = None

    class Config:
        from_attributes = True

class QuizSubmission(BaseModel):
    answers: Dict[str, int]

class QuizResult(BaseModel):
    score: int
    passed: bool
    correctAnswers: int
    totalQuestions: int

class EnrollmentCodeRequest(BaseModel):
    enrollmentCode: str

class EnrollmentResponse(BaseModel):
    success: bool
    message: str

File: front/src/types/course.ts
export interface Quiz {
  id: string;
  question: string;
  options: string[];
  correctOption: number;
}

export interface Chapter {
  id: string;
  title: string;
  content: string;
  quiz: Quiz[];
  completed?: boolean;
}

export interface Course {
  id: string;
  title: string;
  description: string;
  imageUrl: string;
  chapters: Chapter[];
  progress?: number;
  enrolled?: boolean;
  enrollmentCode?: string;
}

export interface QuizResult {
  score: number;
  passed: boolean;
  correctAnswers: number;
  totalQuestions: number;
}
